.github/workflows/lint-codeql.yaml

```
@@ -42,7 +42,7 @@ jobs:
 
   analyze:
     needs: check_changes
-    if: ${{ needs.check_changes.outputs.go-changes == 'true' || github.event_name != 'pull_request' }}
+    if: ${{ github.repository_owner == 'cilium' && github.repository_name == 'cilium' && needs.check_changes.outputs.go-changes == 'true' || github.event_name != 'pull_request' }}
```

This could be simplified using:
```suggestion
    if: ${{ github.repository == 'cilium/cilium' && needs.check_changes.outputs.go-changes == 'true' || github.event_name != 'pull_request' }}
```